lib/src/context/component_wrappers.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:nyxx/nyxx.dart';
+import 'package:nyxx_commands/src/context/context.dart';
+import 'package:nyxx_interactions/nyxx_interactions.dart';
+import 'package:random_string/random_string.dart';
+
+mixin ComponentWrappersMixin implements IContext {
+  @override
+  Future<IMultiselectInteractionEvent> getSelection(MultiselectBuilder selectionMenu,
+          {bool authorOnly = true, Duration? timeout = const Duration(minutes: 12)}) =>
+      commands.interactions.events.onMultiselectEvent
+          .where((event) => event.interaction.customId == selectionMenu.customId)
+          .map((event) => event..acknowledge())
+          .where(
+            (event) =>
+                !authorOnly ||
+                (event.interaction.memberAuthor ?? event.interaction.userAuthor as SnowflakeEntity)
+                        .id ==
+                    user.id,
+          )
+          .timeout(
+            timeout ?? Duration(),
+            onTimeout: timeout != null ? null : (sink) {},
+          )
+          .first;
+
+  @override
+  Future<IButtonInteractionEvent> getButtonPress(Iterable<ButtonBuilder> buttons,
+          {bool authorOnly = true, Duration? timeout = const Duration(minutes: 12)}) =>
+      commands.interactions.events.onButtonEvent
+          .where((event) =>
+              buttons.map((button) => button.customId).contains(event.interaction.customId))
+          .map((event) => event..acknowledge())
+          .where(
+            (event) =>
+                !authorOnly ||
+                (event.interaction.memberAuthor ?? event.interaction.userAuthor as SnowflakeEntity)
+                        .id ==
+                    user.id,
+          )
+          .timeout(
+            timeout ?? Duration(),
```

Comment:
what would happen if timeout is null? timeout is gonna be instant?

Reply:
kinda weird that timeout has default value but its nullable anyway, whats the use case?

Reply:
A null timeout will indicate that the user does not want the interaction awaiting to timeout.

In that case, a default timeout of 0 is passed to the `timeout` method on the stream. However, what matters is the `onTimeout` function provided: if the user-provided timeout is `null`, then an empty function is passed which causes the stream to continue as normal.

However, in most cases users will want to respond using the current interaction, so a timeout is necessary that avoid token expiration. If the user wants no timeout, they have to be explicit and manually set `timeout` to null.